src/operators/publish.ts

```diff
@@ -20,7 +21,7 @@ export function publish<T>(selector: MonoTypeOperatorFunction<T>): MonoTypeOpera
  * @method publish
  * @owner Observable
  */
-export function publish<T>(selector?: MonoTypeOperatorFunction<T>): MonoTypeOperatorFunction<T> {
+export function publish<T, R>(selector?: OperatorFunction<T, R>): MonoTypeOperatorFunction<T> | OperatorFunction<T, R> {
   return selector ?
```

Comment:
I think you can go with just OperatorFunction<T, R> as the return type.

Reply:
Nope. You can't. That will effect the error I mentioned in response to your initial change request.

Reply:
It a necessity. I don't see it as a problem, as TypeScript uses only the overload signatures when matching methods calls. It won't match calls against the implementation signature - which I why I have the of whacking implementation signatures with the `any` hammer. That is, the implementation signature does not form part of the method's public contract; it only enforces the internal contract.

Reply:
Yup. I see why now... This works. Sorry for any confusion.